Version error detection breaking CI

e.g. https://travis-ci.org/WebAssembly/binaryen/jobs/641013614?utm_medium=notification&utm_source=github_status

```
fatal: No names found, cannot describe anything.
CMake Error at CMakeLists.txt:25 (message):
  Error running git describe to determine version
```
(example PR: https://github.com/WebAssembly/binaryen/pull/2616)

cc @sbc100

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
                GIT_VERSION
            OUTPUT_STRIP_TRAILING_WHITESPACE)
   if(${GIT_VERSION_RESULT})
-    message(FATAL_ERROR "Error running git describe to determine version")
+    message(WARNING "Error running git describe to determine version")
   else()
     set(CMAKE_PROJECT_VERSION "${CMAKE_PROJECT_VERSION} (${GIT_VERSION})")
   endif()

